src/workerd/api/streams/internal.c++

```diff
@@ -1520,14 +1562,12 @@ jsg::Promise<void> WritableStreamInternalController::writeLoopAfterFrontOutputLo
         request.source.release(js);
         // If the source is closed, the spec requires us to close the destination unless the
         // preventClose option is true.
-        if (!request.preventClose) {
-          if (!isClosedOrClosing()) {
-            doClose(js);
-          } else {
-            writeState.init<Unlocked>();
-          }
-          return js.resolvedPromise();
+        if (!request.preventClose && !isClosedOrClosing()) {
+          doClose(js);
+        } else {
+          writeState.init<Unlocked>();
         }
+        return js.resolvedPromise();
```

Comment:
This changed block here appears to be a latent bug in the handling of the source being closed. Hitting this is very much an edge case so shouldn't have much impact but definitely worth fixing.